modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/maintenance/CleanFolderAction.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2023 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.cache.persistence.maintenance;
+
+import java.io.File;
+import org.apache.ignite.internal.GridKernalContext;
+import org.apache.ignite.maintenance.MaintenanceAction;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * The action clears storage folders with all data files.
+ */
+public class CleanFolderAction implements MaintenanceAction<Void> {
+    /** */
+    public static final String ACTION_NAME = "clean_forder";
+
+    /** */
+    private final File rootStoreDir;
+
+    /** */
+    private final String[] cacheStoreFolders;
+
+    /** Grid context. */
+    private final GridKernalContext ctx;
```

Comment:
We don't actually need an access to `GridKernalContext` here, reference to `MaintenanceRegistry` instance is enough. Please pass to the constructor registry instance instead of kernal context.